database/seeds/ModelSeeders/BeatmapSeeder.php

```diff
@@ -0,0 +1,139 @@
+
+            return;
+        }
+        $api = '&k='.$api_key;
+
+        try {
+            $beatmaps = json_decode(file_get_contents($base_url.'get_beatmaps?since=2016-01-01%2000:00:00'.$api));
+            $last_beatmapset = null;
+            $beatmap_diff_names = [];
+            $beatmapset_versions = 0;
+            $set_playcount = 0;
+            $number_of_beatmaps = count($beatmaps);
+            $i = 0;
+            $first_map = true;
+            $last_map = false;
+
+            foreach ($beatmaps as $bm) {
+                $make_new_set = false;
+                if ($i === $number_of_beatmaps - 1) {
+                    $make_new_set = true;
+                    $last_map = true;
+                }
+
+                // Here we are going to check if the current beatmap belongs to a new set, and make the set if necessary
+                if ($last_beatmapset === $bm->beatmapset_id || $first_map === true) {
+                    ++$beatmapset_versions;
+                    $beatmap_diff_names[] = $bm->version.'@'.$bm->mode;
+                    $set_playcount += $bm->playcount;
+                } else {
+                    $make_new_set = true;
+                }
+                if ($make_new_set === true) {
+                    if ($last_map === true) {
+                        $the_beatmap = $bm;
+                    } else {
+                        $the_beatmap = $previous_beatmap;
+                    }
+                    // Create new beatmapset
+                    $set = \App\Models\BeatmapSet::where('beatmapset_id', $the_beatmap->beatmapset_id)->first();
+                    if ($set) {
+                        $set->delete();
+                        $overbeatmapsets[] = $the_beatmap->beatmapset_id;
+                    }
+                    $beatmap_diff_names = implode(',', $beatmap_diff_names);
+                    $set = new \App\Models\BeatmapSet;
+                    $set->beatmapset_id = $the_beatmap->beatmapset_id;
+                    $set->creator = $the_beatmap->creator;
+                    $set->artist = $the_beatmap->artist;
+                    $set->title = $the_beatmap->title;
+                    $set->displaytitle = $the_beatmap->title;
+                    $set->source = $the_beatmap->source;
+                    $set->tags = $the_beatmap->tags;
+                    $set->bpm = $the_beatmap->bpm;
+                    $set->approved = $the_beatmap->approved;
+                    $set->approved_date = $the_beatmap->approved_date;
+                    $set->genre_id = $the_beatmap->genre_id;
+                    $set->language_id = $the_beatmap->language_id;
+                    $set->versions_available = $beatmapset_versions;
+                    $set->difficulty_names = $beatmap_diff_names;
+                    $set->play_count = $set_playcount;
+                    $set->favourite_count = $the_beatmap->favourite_count;
+                    $set->save();
```

Comment:
Right, I was wondering whether that field was necessary or not. I suppose I could put in random user IDs after seeding the users?